src/dune_rules/pkg_rules.ml

```diff
@@ -1171,11 +1180,13 @@ module rec Resolve : sig
     :  DB.t
     -> Context_name.t
     -> Loc.t * Package.Name.t
+    -> string list
```

Comment:
Try to avoid arguments with no indication of their purpose in function signatures. Either introduce a type to clarify the intent (e.g. `Components.t = string list`), name the parameter (`~components`) or add a comment.